drivers/src/persistent.rs

```diff
@@ -0,0 +1,127 @@
+// Licensed under the Apache-2.0 license
+
+use core::{marker::PhantomData, mem::size_of, ptr::addr_of};
+
+use caliptra_image_types::ImageManifest;
+use zerocopy::{AsBytes, FromBytes};
+
+use crate::{fuse_log::FuseLogEntry, memory_layout, pcr_log::PcrLogEntry, FirmwareHandoffTable};
+
+#[derive(FromBytes, AsBytes)]
+#[repr(C)]
+pub struct PersistentData {
+    pub manifest1: ImageManifest,
+    reserved0: [u8; memory_layout::MAN1_SIZE as usize - size_of::<ImageManifest>()],
+
+    pub manifest2: ImageManifest,
+    reserved1: [u8; memory_layout::MAN2_SIZE as usize - size_of::<ImageManifest>()],
+
+    pub fht: FirmwareHandoffTable,
+    reserved2: [u8; memory_layout::FHT_SIZE as usize - size_of::<FirmwareHandoffTable>()],
+
+    // TODO: Do we want to hide these fields from the FMC/runtime and force them
+    // to go through the FHT addresses?
```

Comment:
I think that would be a good idea

Reply:
Perhaps, but I want to defer this to a future PR; there's not a super easy way to hide these from some crates and not others.

Also, I'm a bit nervous on relying on the FHT pointers, as if they're corrupted we could wander into undefined behavior land; I'd feel more comfortable if we just added a check that `(&persistent_data.fht.ldevid_tbs_addr) == (&persistent_data.ldevid_tbs as u32)`